libproxy: build failure on darwin

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build nixpkgs#libproxy`

### Build log
```
CMake Error at libproxy/cmake/modules/pacrunner_duktape.cmk:10 (px_check_modules):
  Unknown CMake command "px_check_modules".
Call Stack (most recent call first):
  libproxy/cmake/modules.cmk:21 (include)
  libproxy/CMakeLists.txt:14 (include)


-- Configuring incomplete, errors occurred!
```

See [Hydra](https://hydra.nixos.org/job/nixpkgs/trunk/libproxy.x86_64-darwin).

### Additional context

- Regression of #181917
- Blocks: #157874
- Blocks: porting GSound and GNOME Clocks to macOS
- Notify upstream: libproxy/libproxy#161

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

--- a/pkgs/development/libraries/libproxy/default.nix
+++ b/pkgs/development/libraries/libproxy/default.nix
@@ -1,5 +1,7 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchFromGitHub
+, fetchpatch
 , pkg-config
 , cmake
 , zlib
@@ -27,6 +29,14 @@ stdenv.mkDerivation rec {
     rev = version;
     hash = "sha256-pqj1LwRdOK2CUu3hYIsogQIXxWzShDuKEbDTbtWkgnQ=";
   };
+
+  patches = lib.optionals stdenv.isDarwin [
+    # https://github.com/libproxy/libproxy/pull/189
+    (fetchpatch {
+      url = "https://github.com/libproxy/libproxy/commit/4331b9db427ce2c25ff5eeb597bec4bc35ed1a0b.patch";
+      sha256 = "sha256-uTh3rYVvEke1iWVHsT3Zj2H1F+gyLrffcmyt0JEKaCA=";
+    })
+  ];
 
   outputs = [ "out" "dev" "py3" ];
 

